File: mirrord/layer/tests/apps/issue1123.mjs
import { createServer } from 'net';

console.log("Testing that double binding on same address fails!");
const server = createServer();
server.listen(
  {
    host: "localhost",
    port: 80,
  },
  function () {
    console.log("First bind success!");
    
    const failServer = createServer();
    failServer.listen(
      {
        host: "localhost",
        port: 80,
      },
      function () {
        console.log("Second bind???");
        console.error("Second bind success, but it should fail!");
        exit(-1);
      });

    server.on("error", (fail) => {
      console.log("Second bind???");
      console.error(`First bind failed ${fail}`);
      throw fail;
    });

    failServer.on("error", (_) => {
      console.log(`Second bind failed successfully!`);
      process.exit();
    });
  }
);

File: mirrord/layer/tests/apps/issue1123/issue1123.rs
use std::net::{SocketAddr, TcpListener};

/// Test that double binding on the same address:port combination fails the second time around.
fn main() {
    println!("test issue 1123: START");
    let localhost: SocketAddr = "127.0.0.1:41222".parse().unwrap();
    let _listener_1 = TcpListener::bind(localhost).expect("first bind failed");
    TcpListener::bind(localhost).expect_err("second bind did not fail!");
    println!("test issue 1123: SUCCESS");
}
